run_simulator.sh

```diff
@@ -0,0 +1,179 @@
+#! /usr/bin/sh
+
+VENV="venv"
+BIN=${VENV}"/bin"
+PYTHONPATH="./"
+PYTHON="env PYTHONPATH=$PYTHONPATH $BIN/python"
+PIP=$BIN"/pip"
+PYTEST=$BIN"/pytest"
+TEST_FOLDER=$PYTHONPATH"./tests/"
+
+REQUIREMENTS="-r requirements.txt"
+PRE_COMMIT=$BIN"/pre-commit"
+
+PYAPP="knot_thing_simulator"
+PYMODULE="simulator"
+
+RAW_CONFIG="./config/config.json"
+THING_CONFIG="./config/thing_config.json"
+
+COLOR_GREEN="\e[32m"
+COLOR_RED="\e[31m"
+COLOR_YELLOW="\e[33m"
+STYLE_BOLD="\e[1m"
+STYLE_RST="\e[0m"
+
+helper() {
+  echo "### KNoT Thing Simulator"
+  echo "Usage: $0 [OPTION]... [FILE]"
+  echo " -c,                   $COLOR_GREEN Cleans py cache, py test cache and objects$STYLE_RST"
+  echo " -a,                   $COLOR_GREEN clean-All - performs -c and cleans venv environment$STYLE_RST"
+  echo " -h,                   $COLOR_GREEN Output this helper$STYLE_RST"
+  echo " -r [FILE],            $COLOR_GREEN Runs simulator with a raw config file$STYLE_RST"
+  echo " -t [FILE],            $COLOR_GREEN Runs simulator with custom thing config file$STYLE_RST"
+  echo " Running a custom raw file example:"
+  echo " $0 -r <path-to-config>/raw_config.json"
+}
+
+print_info() {
+  echo "$COLOR_GREEN["$STYLE_BOLD"INFO\e[0m"$COLOR_GREEN"]$STYLE_RST $@"
+}
+
+print_warning() {
+  echo "$COLOR_YELLOW["$STYLE_BOLD"WARNING\e[0m"$COLOR_YELLOW"]$STYLE_RST $@"
+}
+
+print_error() {
+  echo "$COLOR_RED["$STYLE_BOLD"ERRO\e[0m"$COLOR_RED"]$STYLE_RST $@"
+}
+
+create_venv() {
+  print_info "Creating VENV ..."
+  local ret="python3 -m venv $VENV 2> /dev/null"
+  eval $ret
+}
+
+install_requirements() {
+  print_info "Installing $PYMODULE requirements..."
+  local ret="$PIP install $REQUIREMENTS 2> /dev/null"
+  eval $ret
+}
+
+install_precommit_hooks() {
+  print_info "Installing pre-commit hooks..."
+  local cmd="cp hooks/pre-commit .git/hooks/pre-commit 2> /dev/null"
+  eval $cmd
+  local install_pre="$PRE_COMMIT install 2> /dev/null"
+  eval $install_pre
+}
+
+bootstrap() {
+  has_err=$?;create_venv
+  if [ $has_err -ne 0 ]; then
+    print_error "Unable to create env: $STYLE_BOLD$res"
+    exit
+  else
+    print_info "ENV: $VENV created."
+  fi
+  has_err=$?;install_requirements
+  if [ $has_err -ne 0 ]; then
+    clean_all
```

Comment:
Is this necessary?

Reply:
i think it is.. since you couldn't install the requirements, there's no point of preserving the venv